source/reporters/ut_teamcity_reporter.tpb

```diff
@@ -24,84 +24,110 @@ create or replace type body ut_teamcity_reporter is
-
+  
+    l_test_full_name := lower(a_test.item.owner_name) || '.' || lower(a_test.item.object_name) || '.' ||
+                        lower(a_test.item.procedure_name);
+  
+    self.print_text(ut_teamcity_reporter_helper.test_started(a_test_name               => l_test_full_name
+                                                            ,a_capture_standard_output => true));
+  
   end;
 
   overriding member procedure after_calling_test(self in out nocopy ut_teamcity_reporter, a_test in ut_test) is
-    l_assert           ut_assert_result;
-    l_test_full_name   varchar2(4000);
+    l_assert         ut_assert_result;
+    l_test_full_name varchar2(4000);
     procedure print_output(a_exectable ut_executable) is
       l_lines ut_varchar2_list;
     begin
-      if a_exectable is not null and a_exectable.is_defined and a_exectable.serveroutput is not null and dbms_lob.getlength(a_exectable.serveroutput) > 0 then
+      if a_exectable is not null and a_exectable.is_defined and a_exectable.serveroutput is not null and
+         dbms_lob.getlength(a_exectable.serveroutput) > 0 then
         l_lines := ut_utils.clob_to_table(a_exectable.serveroutput);
-        for i in 1..l_lines.count loop
+        for i in 1 .. l_lines.count loop
           self.print_text(l_lines(i));
         end loop;
       end if;
     end;
   begin
---    l_test_full_name := self.suite_names_stack(self.suite_names_stack.last) || ':' ||
---                        nvl(replace(a_test.description, '.'), a_test.name);
-    l_test_full_name := lower(a_test.item.owner_name)||'.'||lower(a_test.item.object_name)||'.'||lower(a_test.item.procedure_name);
-
+    --    l_test_full_name := self.suite_names_stack(self.suite_names_stack.last) || ':' ||
+    --                        nvl(replace(a_test.description, '.'), a_test.name);
+    l_test_full_name := lower(a_test.item.owner_name) || '.' || lower(a_test.item.object_name) || '.' ||
+                        lower(a_test.item.procedure_name);
+  
     if a_test.result = ut_utils.tr_ignore then
       self.print_text(ut_teamcity_reporter_helper.test_ignored(l_test_full_name));
     else
-      
+    
       -- reproduce the output from before/after procedures and the test
       print_output(a_test.before_each);
       print_output(a_test.before_test);
       print_output(a_test.item);
       print_output(a_test.after_test);
       print_output(a_test.after_each);
-
-      if a_test.results is not null and a_test.results.count > 0 then
+    
+      if a_test.result = ut_utils.tr_error then
+        self.print_text(ut_teamcity_reporter_helper.test_std_err(a_test_name => l_test_full_name
+                                                                ,a_out       => coalesce(a_test.before_each.error_backtrace
+                                                                                        ,a_test.before_test.error_backtrace
```

Comment:
If both `before_test` and `after_test` will throw, then we will not show the stacktrace from the after block.